src/Features/CSharp/Portable/CodeRefactorings/InlineMethod/CSharpInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,244 @@
+        }
+
+        private static SyntaxNode? GetExpressionFromStatementSyntaxNode(StatementSyntax statementSyntax)
+            => statementSyntax switch
+            {
+                ReturnStatementSyntax returnStatementSyntax => returnStatementSyntax.Expression,
+                ExpressionStatementSyntax expressionStatementSyntax => expressionStatementSyntax.Expression,
+                _ => null
+            };
+
+        protected override bool ShouldCheckTheExpressionPrecedenceInCallee(SyntaxNode syntaxNode)
+            => s_syntaxKindsNeedsToCheckThePrecedence.Any(syntaxNode.IsKind);
+
+        protected override bool NeedWrapInParenthesisWhenPrecedenceAreEqual(SyntaxNode calleeInvocationSyntaxNode)
+        {
+            var parent = calleeInvocationSyntaxNode.Parent;
+            if (parent != null)
+            {
+                // For left associative expression
+                // If in the original invocation, it is the left child. Since it is a left associative expression,
+                // then it is safe to replace it without parenthesis.
+                // Example:
+                // int Callee(int i, int j) => i + j;
+                // void Caller()
+                // {
+                //   var a = F(1, 2) - 3 - 2 - 1;
+                // }
+                if (s_leftAssociativeSyntaxKinds.Any(parent.IsKind) && parent is BinaryExpressionSyntax leftAssociativeBinaryExpressionSyntax)
+                {
+                    return leftAssociativeBinaryExpressionSyntax.Left != calleeInvocationSyntaxNode;
+                }
+
+                // Same for right associative expression, if the original invocation is the right child,
+                // it is safe to replace it.
+                if (parent.IsKind(SyntaxKind.CoalesceExpression)
+                    && parent is BinaryExpressionSyntax rightAssociativeBinaryExpressionSyntax)
+                {
+                    return rightAssociativeBinaryExpressionSyntax.Right != calleeInvocationSyntaxNode;
+                }
+                else if (parent is ConditionalExpressionSyntax conditionalExpressionSyntax)
+                {
+                    return conditionalExpressionSyntax.WhenFalse != calleeInvocationSyntaxNode;
+                }
+                else if (parent is AssignmentExpressionSyntax assignmentExpressionSyntax)
+                {
+                    return assignmentExpressionSyntax.Right != calleeInvocationSyntaxNode;
+                }
+            }
+
+            // In other cases, always put a 'safe' parenthesis around
+            return true;
+        }
+
+        protected override SyntaxNode GenerateTypeSyntax(ITypeSymbol symbol)
+            => symbol.GenerateTypeSyntax(allowVar: false);
+
+        protected override SyntaxNode GenerateArrayInitializerExpression(ImmutableArray<SyntaxNode> arguments)
+            => SyntaxFactory.InitializerExpression(SyntaxKind.ArrayInitializerExpression, SyntaxFactory.SeparatedList(arguments));
```

Comment:
consider moving to SyntaxGenerator.

Reply:
Actually SyntaxGenerator has new Array helper method. but I found that it has an error so I logged a bug for that https://github.com/dotnet/roslyn/issues/46651
I am preferring the switch to use that one once the bug is resolved. :)